### ISSUE TITLE
CancellationToken ThrowIfCancellationRequested exceptions in documentation seems incorrect

### ISSUE THREAD

**Issue by mkwaner** [dotnet/runtime#62854]
The documentation for `CancellationToken.ThrowIfCancellationRequested` indicates an `ObjectDisposedException` will be thrown if "The associated CancellationTokenSource has been disposed." This doesn't seem correct from inspecting the source code. Looks like it just calls `IsCancellationRequested` before throwing `OperationCanceledException` and `IsCancellationRequested` wouldn't throw `ObjectDisposedException` either when tracing it through to `CancellationTokenSource`.

**Comment by buyaa-n** [comment]
You are right, I guess the intention was to throw `ObjectDisposedException`, and I believe throwing it now would be breaking change, probably we would just change the doc instead if @stephentoub not mind
